utils/osm_incremental_cagg_refresh/README.md

```diff
@@ -0,0 +1,107 @@
+        schedule_interval => '5 seconds',
+        config => '{"enable_tiered_reads": true, "max_runtime": "5 minutes"}'
+    );
+```
+
+This call adds a job that runs in the background and refreshes the continuous aggregate.
+
+**NOTE**
+The `enable_tiered_reads = true` config is necessary if the default DB settings for `timescaleb_osm.enable_tiered_reads` GUC is `false`.
+
+### 3. Monitoring
+
+The `_timescaledb_additional.osm_incremental_refresh_status` view can be used to monitor progress of the refresh execution:
+```sql
+tsdb=> SELECT * FROM _timescaledb_additional.osm_incremental_refresh_status;
+       continuous_aggregate       | not started | started | finished 
+----------------------------------+-------------+---------+----------
+ hf_schema.cagg_data_1minute      |         256 |       0 |        3
+```
+
+`finished` shows the number of refresh time intervals that have been completed, `not started` shows how many are remaining.
+
+The `_timescaledb_additional.job_cagg_refresh_status` view can be used to check the status of the job executions (consumers):
+```sql
+tsdb=> SELECT * FROM _timescaledb_additional.job_cagg_refresh_status;
+    clock_timestamp     |  pid   |  wait_event  |                        application_name                        | xact_age | backend_age 
+------------------------+--------+--------------+----------------------------------------------------------------+----------+-------------
+ 2025-01-04 14:31:03+00 | 153087 |              | hf_schema.cagg_data_1minute refresh 2021-01-19 2021-02-18    | 00:00:10 | 00:00:44
+```
+
+## Advanced Usecase
+
+1. You can use the schedule_osm_cagg_refresh procedure to generate refresh ranges for multiple continuous aggregates.
+
+```sql
+CALL _timescaledb_additional.schedule_osm_cagg_refresh(); -- All CAggs with tiered data
+```
+
+```sql
+CALL _timescaledb_additional.schedule_osm_cagg_refresh(schema_mask=>'hf_schema'); -- All CAggs with tiered data under hf_schema 
+```
+
+```sql
+CALL _timescaledb_additional.schedule_osm_cagg_refresh( name_mask=>'cagg_data%'); -- All CAggs with name beginning with cagg_data, that appears in any schema  
+```
+
+```sql
+CALL _timescaledb_additional.schedule_osm_cagg_refresh( schema_mask=>'hf_schema%', name_mask=>'cagg_data%'); -- All CAggs with name beginning with cagg_data under schemas that begin with the name hf_schema.  
+```
+
+2. Multiple continuous aggregates can be refreshed in parallel. You can create multiple jobs like so:
+
+```sql
+SELECT
+    add_job(
+        '_timescaledb_additional.task_refresh_continuous_aggregate_incremental_runner',
+        schedule_interval => '5 seconds',
+        config => '{"enable_tiered_reads": true, "max_runtime": "5 minutes"}'
+    )
+FROM
+    generate_series(1, 4);
+```
+
+Every job is run by a background worker. Please keep you max background worker settings and database load in mind while adding jobs so that you do not exhaust DB resources.
```

Comment:
```suggestion
Every job is executed by a background worker. Please keep you max background worker settings and database load in mind while adding jobs so that you do not exhaust DB resources.
```